Design-Docs/standardizing-cluster-infrastructure.md

```diff
@@ -0,0 +1,277 @@
+
+Every cloud provider has to have a Cluster API provider integration and a cloud controller manager (CCM). It can be both an “actual” cloud provider with API and a bare-metal system (based on OpenStack).
+
+To facilitate the configuration of applications in these clusters, we propose to standardize CCM (load balancer) and naming of storage classes, so that applications can be configured independently of the cluster infrastructure.  
+
+## Motivation
+
+The main goal of SCS is to ensure the users’ and cluster operators’ sovereignty while using European cloud infrastructure. With this proposal, we want to ensure that the infrastructure, from the node image to core Kubernetes components, is transparent in all SCS providers. We want to have a concise structure and good documentation on what providers have to do to get certified by SCS and aim to keep the certification process itself lean.
+
+Furthermore, we consider the freedom of choosing a provider a key to the sovereignty of cluster operators. They should operate their clusters fully independently of the provider infrastructure.
+
+For users who launch and run applications on SCS-certified infrastructure, we want to ensure that they do not have to worry about security issues. SCS infrastructure should only get certified when all aspects of security are taken care of. The users should be able to configure their applications independently of the cluster infrastructure. Only this will lead to actual sovereignty, as they would otherwise have to worry about certain security issues, e.g. related to metrics-server with self-signed certificates.
+
+### Goals
+
+- Standardize infrastructure (e.g. definition of load balancers) of European cloud providers
+- Easy certification process for cloud providers
+- Concise documentation on how to fulfill SCS certification
+- Applications are configured independently of the cloud provider
+- Automated e2e tests for cluster and application configuration
+- Secure cluster infrastructure for users and cluster operators
+- Auditability & automated testing (unit and e2e tests) of SCS code
+
+### Non-goals
+
+- This proposal does not intend to define specifically all inputs, e.g. to define load balancers in SCS infrastructure. This is left for an open discussion in the SCS team and with cloud providers.
+- We don’t want to say that any specific components are must-haves for cluster infrastructure, but we will give suggestions.
+
+## Proposal
+
+This proposal does not specify any details about load balancer definitions, node images, or server types. It rather aims to define a way to handle standardization and certification in general.
+
+It consists of three parts:
+
+### Proposal 1: Pre-defined cluster stacks are necessary to ensure that clusters and applications work - a free configuration of cluster components is not possible
```

Comment:
`s/Proposal/Part/`, I initially read these as mutually exclusive, but I don't think now that's how it was intended?

Reply:
@horazont no, they are not exclusive. They are, more or less, independent and all necessary to achieve the independence of layers (such as applications independent of infrastructure provider)